### PR TITLE
Avoid errors from missing authorship in array (fix for #25)

### PR DESCRIPTION
**Pull request by sebastiangreger**

### CODE REVIEW

File: classes/Cron.php [hunk 1]
```diff
@@ -146,27 +146,48 @@ public static function parseWebmention($request)
...
                     }
-                    if ($result['author']['url'] == ''  && isset($mf2item['properties']['url'][0])) {
+                    if (empty($result['author']['url']) && !empty($mf2item['properties']['url'][0])) {
                         $result['author']['url'] = $mf2item['properties']['url'][0];
                     }
                 }
             }
 
+            // if no h-card was found, try to use 'author' property of h-entry instead
+            if (!$hcardfound ?? false) {
```
Comment (fabianmichael) [inline classes/Cron.php:179]: This is quite elegant, but very hard to read … is would be easier to understand, if you set `$hcardfound = false` before the foreach-loop and remove the ` ?? false` part.

File: classes/Cron.php [hunk 2]
```diff
@@ -236,9 +257,9 @@ public static function parseWebmention($request)
             // create the commention data
             $finaldata = [
                 'status' => Commentions::defaultstatus($result['type']),
-                'name' => $result['author']['name'],
-                'website' => $result['author']['url'],
-                'avatar' => $result['author']['photo'],
+                'name' => $result['author']['name'] ?? null,
+                'website' => $result['author']['url'] ?? null,
+                'avatar' => $result['author']['photo'] ?? null,
```
Comment (fabianmichael) [inline classes/Cron.php:262]: The `php-comments`lib returns `false` for each of these values by default (see https://github.com/indieweb/php-comments/blob/master/src/indieweb/comments.php#L66), if they’re empty. As far as I understand, if these values are missing from the webmention source, the returned value will be `false`. But if the `url` field does not validate and thus is unset in line 164, its value in the `$finaldata` array would be `null`. Am I right with this?

File: classes/Cron.php (hunk 1 above)
Comment (sebastiangreger) [reply classes/Cron.php:179]: Agreed. I'll make a mental note and clean this up later - I want to revisit this flow one more time anyways...

File: classes/Cron.php (hunk 2 above)
Comment (sebastiangreger) [reply classes/Cron.php]: Yes, this is inconsistent indeed. The outcome should be the same as `Commentions::sanitize()` checks for `empty()` (so false or null lead to the same result), but I cleaned it up and updated the PR.

**Comment by sebastiangreger** [comment]
Yes, welcome to the circus that is microformats parsing :clown_face: 

I am going to revisit this entire flow in the near future (though not necessarily before 1.0) - it grew as I figured out one aspect at a time and could benefit from a thorough cleanup.

**Comment by fabianmichael** [comment]
@sebastiangreger Yes, things like this always get to a mindblowing complexity real fast … I would prefer to use `null` instead of false, as `null` really means not value and would be much more consistent, as `false` is indeed a non-empty value. But if you want to revisit this anyways, we can leave it like that for now. Merging this for now. :-)

**Pull request merged**